cli_tools_tests/e2e/gce_ovf_import/test_suites/ovf_instance_import/ovf_instance_import_test_suite.go

```diff
@@ -170,31 +177,21 @@ func deleteImagesIfInstanceCreationFails(ctx context.Context, testCase *junitxml
 	//
 	// One goroutine is started per image, and it polls for the temporary image that
 	// we're expecting to see. Once seen, it quits.
-	detectedImages := map[string]bool{}
-	for i := 1; i < 4; i++ {
-		imgName := fmt.Sprintf("ovf-%s-%d", buildID, i)
-		detectedImages[imgName] = false
-		go func() {
-			for {
-				image, _ := client.GetImage(testProjectConfig.TestProjectID, imgName)
-				if image != nil {
-					detectedImages[imgName] = true
-					return
-				}
-				time.Sleep(10 * time.Second)
-			}
-		}()
-	}
 	e2e.RunTestCommandAssertErrorMessage(cmds[testType], buildTestArgs(props, testProjectConfig)[testType], "not-a-machine-type' was not found", logger, testCase)
 
-	// Check that the expected images were seen during import and that they were deleted.
-	for imgName, detected := range detectedImages {
-		if !detected {
-			e2e.Failure(testCase, logger, fmt.Sprintf("Didn't see expected image %s", imgName))
-		}
-		image, _ := client.GetImage(testProjectConfig.TestProjectID, imgName)
-		if image != nil {
-			e2e.Failure(testCase, logger, fmt.Sprintf("Expected image %s to be removed", imgName))
+	// check if boot image is deleted
+	imgName := fmt.Sprintf("%s-boot-image", props.instanceName)
+	image, _ := client.GetImage(testProjectConfig.TestProjectID, imgName)
+	if image != nil {
+		e2e.Failure(testCase, logger, fmt.Sprintf("Expected image %s to be removed", imgName))
+	}
+
+	// check if data disks are deleted
+	for i := 0; i < 2; i++ {
+		diskName := daisyovfutils.GenerateDataDiskName(props.instanceName, i)
+		disk, _ := client.GetDisk(testProjectConfig.TestProjectID, testProjectConfig.TestZone, diskName)
+		if disk != nil {
+			e2e.Failure(testCase, logger, fmt.Sprintf("Expected disk %s to be removed", diskName))
 		}
```

Comment:
You're right, it's a mistake it should be i+1 instead of i, but it pass as it's search for disk-0 & disk-1 and don't don't find it which is expected. but I'll fix it to be i + 1.
regarding the boot image, I already checked the deletion of it, check line 182

Reply:
Thank you for fixing!

Regarding the boot image, I saw that you checked for the deletion of the image, I am just curious if we need to check for the deletion of the boot **disk**. What do you think?

Reply:
in OVF-Import, the boot disk is created on the final instance, and in this test the final instance will not be created as the machine type is not correct.

Reply:
It's exactly what I ask, got it, thank you for clarifying!